subprojects/core/src/integTest/groovy/org/gradle/api/tasks/TaskPropertiesRegressionIntegrationTest.groovy

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.tasks
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.util.internal.ToBeImplemented
+import spock.lang.Issue
+
+/**
+ * Tests for previously existing mistakes in task properties.
+ */
+class TaskPropertiesRegressionIntegrationTest extends AbstractIntegrationSpec {
+    @ToBeImplemented
+    @Issue("https://github.com/gradle/gradle/issues/24747")
+    def "nested property with final getter carries task dependencies in nested object"() {
+        given:
+        buildFile("""
+            interface Parameters {
+                @OutputFile
+                RegularFileProperty getOutputFile()
+            }
+
+            abstract class ParameterizedTask<P> extends DefaultTask {
+                private final P params
+
+                protected ParameterizedTask(Class<P> paramType) {
+                    this.params = objects.newInstance(paramType)
+                }
+
+                @Nested
+                final P getParameters() {
```

Comment:
Does this work if `getParameters` is abstract and not generic?

Reply:
It should, yes.